### DISCUSSION TITLE
AT28C64(B) troubles / Apple II europlus

### DISCUSSION THREAD

**Discussion by hanshuebner** [tjboldt/Apple2-IO-RPi#81] [General]
Hi,

first off, thank you for creating this project!  I've built myself a board today to give it a try, and I had mixed results which are probably only partly related to the project itself, though:

As my regular suppliers did not have the AT28C64B in stock or wanted insane amounts of money (13 Euros per piece), I ordered a bunch of chips from AliExpress, knowing that this is a bit of a gamble.  I received an uneven collection of AT28C64's and AT28C64B's, only one of the latter working.  The AT28C64's don't work at all - With some of them, I see some characters of the firmware message, some just don't work at all.  It looks a bit like a speed problem, but as most other cards use much slower EPROM's, I have trouble believing that the read access times are at fault.  I burned the firmware onto an M27C64 to give that a try, but with that,the machine just crashes, too.

This is with an Apple IIe.   I also tried the board in a Generic Taiwanese Apple Clone (GTAC-2), which as far as I understand tries to resemble an Apple II europlus.  If I try to boot from the card using PR#6, it just hangs.  That machine has been difficult in other respects, too, so maybe it is a general compatibility issue.

Any thoughts?  As you can tell, I'm an Apple II newbie, so excuse my ignorance.

Kind regards,
Hans

**Comment by tjboldt** [comment]
Hi Hans,

There are lots of things that can go wrong assembling a card. Check the soldering, note that the orientation of the AT28C64B is opposite of the other chips, the data buffer chip won’t work if you used a 74LVC245 instead of a 74LS245… As for using a 27C64 instead, a few wiring changes are necessary. You would need to jumper power from pin 28 to pin 1 (VCC to VPP). You would probably need to cut the trace off of pin 27 and jumper power from pin 28 to pin 27 (VCC to PGM).

Regards,
Terence

**Reply by tjboldt** [reply]
It’s a little late now that you already had boards made and soldered components but the fourth prototype was functionally equivalent and had jumpers to choose between EEPROM and EPROM. 

https://github.com/tjboldt/Apple2-IO-RPi/tree/6b304ecc3d602ddcdde19a20e6a61731b6903176/Hardware

**Reply by hanshuebner** [reply]
I made the modification (cut trace from EPROM Pin 27, connect 1, 27 and 28), but that did not make it work.  I'm consistently dropped to the monitor with this (C0638- A=01 X=60 Y=FF P=74 S=7B, with some variation in S but the address almost always being the same).  Any ideas?

**Comment by hanshuebner** [comment]
Hi Terence,

I've bought the TTL chips from Mouser - even the LS ones - to make sure that I don't have to deal with untrusted TTL chips screwing up my boards.  I've also built a second board yesterday just to verify that it is not a soldering issue I may have overlooked.  Both boards behave the same with the single working AT28C64B that I have.  In the IIe, things are mostly working (occasionally, I end up in the monitor when I start the system, or I see it hang), the clone machine just hangs when I try to boot using PR#6.  Should the board work in a II (euro)plus compatible machine?  Can I help diagnosing the issue?  I have some knowledge of 6502 assembly, the monitor and troubleshooting hardware.

Thank you for pointing out the pinout differences between the AC28C64B and EPROMs.  I'll rewire one of the boards to see how it behaves with the EPROM that I have burnt.

Thanks again!
Hans

**Comment by hanshuebner** [comment]
I found my problem:  I was not looking at the silk screen carefully enough and populated U0 with a 74LVC245 instead of a 74LS245.  Also, I could get a M27C64 to work after having made the suggested modification.  AT28C64's do not work for me at all.

The two working boards that I have work both in the IIe and the GTAC-2, so all my issues are solved. Thank you for reading!
-Hans

**Reply by tjboldt** [reply]
I'm glad you got it working. I tried one board with the LVC245 in place of the LS245 and it failed in a similar way. The reason for this is the LVC245 doesn't see 3.3V inputs as high values when running at 5V VCC. The LS245 is more tolerant of lower input voltages.
